### PR TITLE
Remove mozlog dependency

### LINKED ISSUE #5: Log type request.summary
I want to use `mozlogrus` to output a request summary.

A stated on the docs, https://wiki.mozilla.org/Firefox/Services/Logging, the `Type` should be set to `request.summary`.

The type seems to be hard-coded to `"app.log"` here.

I'd be happy to contribute something, but what would be the best way to specify it? Use `entry.Data["Type"]` if set?

Thanks

### PR DESCRIPTION
**Pull request by oremj**
Fixes #5

### CODE REVIEW

File: logrus.go [hunk 1]
```diff
@@ -40,12 +38,12 @@ type MozLogFormatter struct {
...
-	appLog := &mozlog.AppLog{
+	appLog := &AppLog{
 		Timestamp:  entry.Time.UTC().UnixNano(),
 		Time:       entry.Time.UTC().Format(time.RFC3339),
 		Type:       "app.log",
 		Logger:     m.LoggerName,
-		Hostname:   mozlog.Hostname(),
+		Hostname:   Hostname(),
 		EnvVersion: "2.0",
 		Pid:        os.Getpid(),
```
Comment (mostlygeek) [inline logrus.go:48]: should probably cache is variable as well to avoid the system call.

File: applog.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	Logger     string
+	Hostname   string `json:",omitempty"`
+	EnvVersion string
+	Pid        int `json:",omitempty"`
+	Severity   int `json:",omitempty"`
+	Fields     map[string]interface{}
+}
+
+// NewAppLog returns a loggable struct
+func NewAppLog(loggerName string, msg []byte) *AppLog {
```
Comment (mostlygeek) [inline applog.go]: add `type string` to the params. It won't break any previous code.

File: logrus.go (hunk 1 above)
Comment (oremj) [reply logrus.go:48]: It calls https://github.com/mozilla-services/go-mozlogrus/pull/6/files#diff-e49fa914308349fb61e3d6ef6b8a0fdeR10 which caches the hostname.

Comment (oremj) [reply logrus.go:48]: I could also add it the hostname caching to an init instead.

File: applog.go (hunk 2 above)
Comment (oremj) [reply applog.go]: added

File: logrus.go [hunk 3]
```diff
@@ -40,12 +38,12 @@ type MozLogFormatter struct {
 }
 
 func (m *MozLogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
-	appLog := &mozlog.AppLog{
+	appLog := &AppLog{
```
Comment (leplatrem) [inline logrus.go:41]: Why not use `NewAppLog()` ?

**Comment by mostlygeek** [comment]
hey @oremj i'm working on a V2 with a less side effect prone API: https://github.com/mozilla-services/go-mozlogrus/tree/v2.  I'm basing it off of this branch.

File: applog.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+// NewAppLog returns a loggable struct
+func NewAppLog(loggerName string, logType string, msg []byte) *AppLog {
+	now := time.Now().UTC()
+	return &AppLog{
+		Timestamp:  now.UnixNano(),
+		Time:       now.Format(time.RFC3339),
+		Type:       logType,
+		Logger:     loggerName,
+		Hostname:   hostname,
```
Comment (leplatrem) [inline applog.go:31]: I believe it would be clearer to use `Hostname()` instead of the cache variable directly :)

**Comment by oremj** [comment]
Superseded by #7

**Pull request closed without merging**